multicluster/controllers/multicluster/leader_clusterset_controller.go

```diff
@@ -156,3 +209,68 @@ func validateLocalClusterClaim(c client.Client, clusterSet *multiclusterv1alpha1
 
 	return
 }
+
+func (r *LeaderClusterSetReconciler) updateStatus() {
+	defer r.mutex.Unlock()
+	r.mutex.Lock()
+
+	if r.clusterSetConfig == nil {
+		// Nothing to do.
+		return
+	}
+
+	status := multiclusterv1alpha1.ClusterSetStatus{}
+	status.TotalClusters = int32(len(r.clusterSetConfig.Spec.Members) + len(r.clusterSetConfig.Spec.Leaders))
+	status.ObservedGeneration = r.clusterSetConfig.Generation
+	clusterStatues := r.StatusManager.GetMemberClusterStatues()
+	status.ClusterStatuses = clusterStatues
+	readyClusters := 0
+	unknownClusters := 0
+	for _, cluster := range clusterStatues {
+		for _, condition := range cluster.Conditions {
+			if condition.Type == multiclusterv1alpha1.ClusterReady {
+				switch condition.Status {
+				case v1.ConditionTrue:
+					readyClusters += 1
+				case v1.ConditionUnknown:
+					unknownClusters += 1
+				}
+			}
+		}
+	}
+	status.ReadyClusters = int32(readyClusters)
+	overallCondition := multiclusterv1alpha1.ClusterSetCondition{
+		Type:               multiclusterv1alpha1.ClusterSetReady,
+		Status:             v1.ConditionFalse,
+		Message:            "",
+		Reason:             NoReadyCluster,
+		LastTransitionTime: metav1.Now(),
+	}
+	if readyClusters == len(r.clusterSetConfig.Spec.Members) {
+		overallCondition.Status = v1.ConditionTrue
+		overallCondition.Reason = ""
```

Comment:
maybe add a reason like `ClusterSet is healthy` to indicate all clusters are healthy?